subsys/mpsl/pin_debug/mpsl_pin_debug_nrf54h.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device_runtime.h>
+#include <mpsl_dppi_protocol_api.h>
+#include <nrfx_gpiote.h>
+#include <nrfx_dppi.h>
+#include <hal/nrf_radio.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <uicr/uicr.h>
+
+static nrfx_gpiote_t gpiote = NRFX_GPIOTE_INSTANCE(0);
+LOG_MODULE_REGISTER(mpsl_radio_pin_debug, CONFIG_MPSL_LOG_LEVEL);
+
+static int m_ppi_config(void)
+{
+	nrf_gpiote_subscribe_set(
+		gpiote.p_reg,
+		nrfx_gpiote_set_task_address_get(
+			&gpiote, CONFIG_MPSL_PIN_DEBUG_RADIO_READY_AND_DISABLED_PIN),
+		MPSL_DPPI_RADIO_PUBLISH_READY_CHANNEL_IDX);
+
+	nrf_gpiote_subscribe_set(
+		gpiote.p_reg,
+		nrfx_gpiote_clr_task_address_get(
+			&gpiote, CONFIG_MPSL_PIN_DEBUG_RADIO_READY_AND_DISABLED_PIN),
+		MPSL_DPPI_RADIO_PUBLISH_DISABLED_CH_IDX);
+
+	nrf_gpiote_subscribe_set(gpiote.p_reg,
+				 nrfx_gpiote_set_task_address_get(
+					 &gpiote, CONFIG_MPSL_PIN_DEBUG_RADIO_ADDRESS_AND_END_PIN),
+				 MPSL_DPPI_RADIO_PUBLISH_ADDRESS_CHANNEL_IDX);
+
+	nrf_gpiote_subscribe_set(gpiote.p_reg,
+				 nrfx_gpiote_clr_task_address_get(
+					 &gpiote, CONFIG_MPSL_PIN_DEBUG_RADIO_ADDRESS_AND_END_PIN),
+				 MPSL_DPPI_RADIO_PUBLISH_END_CHANNEL_IDX);
+
+	return 0;
+}
+
+/* SPU131: P1.4 permissions */
+UICR_SPU_FEATURE_GPIO_PIN_SET(0x5f920000UL, 1, 4, true, NRF_OWNER_RADIOCORE);
+/* SPU131: P1.5 permissions */
+UICR_SPU_FEATURE_GPIO_PIN_SET(0x5f920000UL, 1, 5, true, NRF_OWNER_RADIOCORE);
+/* gpio1 - P1.4 CTRLSEL = 4 */
+UICR_GPIO_PIN_CNF_CTRLSEL_SET(0x5F938200UL, 4, 6);
+/* gpio1 - P1.5 CTRLSEL = 4 */
+UICR_GPIO_PIN_CNF_CTRLSEL_SET(0x5F938200UL, 5, 6);
+
+static int mpsl_radio_pin_debug_init(void)
+{
+	if (IS_ENABLED(CONFIG_PM_DEVICE)) {
+		const struct device *gpio1 = DEVICE_DT_GET(DT_NODELABEL(gpio1));
+
+		int ret = pm_device_runtime_get(gpio1);
+
+		if (ret) {
+			LOG_ERR("Failed to get runtime for GPIO: %d", ret);
+			return ret;
+		}
+	}
+
+	/* GPIOTE channel allocation is not available for GPIOTE0. Simply use channel 0 and 1 */
+	uint8_t radio_ready_radio_disabled_gpiote_channel = 0;
+	uint8_t radio_address_radio_end_gpiote_channel = 1;
```

Comment:
These variables are used once and most likely will be removed during compilers optimization. Do we need them ?

Reply:
I added them to make the code easier to read. It documents what we are using the values for

Reply:
Then, I suggest to make them constant values. Also I suggest to state that `GPIOTE0` channels are not selected arbitrarily but must mach to `GPIO P1` pins configured in [Line 49-56]( https://github.com/nrfconnect/sdk-nrf/pull/25122/files#diff-aa731e96a44ccf244b92ed74c187a36ce63bd00c9829c2c76fd581a796fb3556R49-R56)